sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1096,8 +1099,42 @@ class SessionCatalog(
    * This performs reflection to decide what type of [[Expression]] to return in the builder.
    */
   protected def makeFunctionBuilder(name: String, functionClassName: String): FunctionBuilder = {
```

Comment:
The changes [here](https://github.com/apache/spark/pull/18700/files#diff-ca4533edbf148c89cc0c564ab6b0aeaa) are for `HiveSessionCatalog`. Also, we have a test case in `HiveUDAFSuite.scala` to verify it.